src/plugins/skills/src/systems/flush_skill_combos.rs

```diff
@@ -0,0 +1,327 @@
+use crate::{
+	skills::QueuedSkill,
+	traits::{Flush, IsTimedOut},
+};
+use bevy::prelude::*;
+use common::traits::{iterate::Iterate, update_cumulative::CumulativeUpdate};
+use std::time::Duration;
+
+pub(crate) fn flush_skill_combos<TCombos, TComboTimeout, TTime, TQueue>(
+	mut agents: Query<(&mut TCombos, Option<&mut TComboTimeout>, &TQueue)>,
+	time: Res<Time<TTime>>,
+) where
+	TCombos: Flush + Component,
+	TComboTimeout: CumulativeUpdate<Duration> + IsTimedOut + Flush + Component,
+	TTime: Default + Sync + Send + 'static,
+	TQueue: Iterate<QueuedSkill> + Component,
+{
+	let delta = time.delta();
+
+	for (mut combos, timeout, queue) in &mut agents {
+		match (skills_queued(queue), timeout) {
+			(true, Some(mut timeout)) => timeout.flush(),
+			(false, None) => combos.flush(),
+			(false, Some(timeout)) => flush_when_timed_out(combos, timeout, delta),
+			_ => {}
+		}
+	}
```

Comment:
If skills are queued and we have no timeout component, we do nothing flush the combos component, because our combo is still active as long as skills are queued and not timed out (which they never are, if there is no timeout component). So the logic seems fine to me.